### ISSUE TITLE
Error on displaying text in any RTL language

### ISSUE THREAD

**Issue by Muaath5** [telegramdesktop/tdesktop#10544]
## Steps to reproduce
1. Use link https://t.me/setlanguage/ar-beta To set language `ar-bata`

## Expected behaviour
Numbers displayed correctly.
English text is displayed correctly.
RTL View for the application (Menu in the right)

### Numbers are reversed!! For example:
1. Chat members count are reversed
2. Poll voters count (also in the tip)
3. The number of the version of the app.

<details>
<summary><b>EXAMPLE (Red means error, Green means Good)</b>:</summary>

![Reversed numbers example](https://user-images.githubusercontent.com/60256639/111162105-e3580e00-85ac-11eb-8325-aba642f890bd.png)

</details>

<details>
<summary><b>ANOTHER EXAMPLE (App version is reversed, It should be 2.6.2, But it's 6.2.2!!)</b>:</summary>

![App version number is reversed](https://user-images.githubusercontent.com/60256639/111161798-91af8380-85ac-11eb-9373-e8c994512e50.png)

</details>

### Some text has wrong alignment! For example:
<details>
<summary><b>EXAMPLE</b>:</summary>

![Wrong alignment example](https://user-images.githubusercontent.com/60256639/111159761-8196a480-85aa-11eb-81eb-e2870b791318.png)

</details>

### Some english text are reversed in displaying!!
<details>
<summary><b>EXAMPLE</b>:</summary>

!["HTTP" word is reversed in Proxy settings](https://user-images.githubusercontent.com/60256639/111160406-1ef1d880-85ab-11eb-8632-ec56837a85c8.png)

</details>

### Configuration
**Operating system:** Microsoft Windows 10 Pro (2004)

**Version of Telegram Desktop:** 2.6.2 (Last version)

**Installation source (Linux Only)** - the official website / GitHub releases / flatpak / snap / distribution package:

**Used theme**: [TgDarkBlueTheme](https://t.me/addtheme/TgDarkBlueTheme)

<details>
<summary><b>Logs</b>:</summary>
[2021.03.15 16:19:38] Launched version: 2006002, install beta: [TRUE], alpha: 0, debug mode: [FALSE]

[2021.03.15 16:19:38] Executable dir: E:/Telegram Desktop/, name: Telegram.exe
[2021.03.15 16:19:38] Initial working dir: E:/Telegram Desktop/
[2021.03.15 16:19:38] Working dir: E:/Telegram Desktop/
[2021.03.15 16:19:38] Command line: E:\Telegram Desktop\Telegram.exe -noupdate -tosettings
[2021.03.15 16:19:38] Executable path before check: E:/Telegram Desktop/Telegram.exe

[2021.03.15 16:19:38] Logs started

[2021.03.15 16:19:38] Connecting local socket to Global\1e941763e028d44b1e269790f00fa357-{87A94AB0-E370-4cde-98D3-ACC110C5967D}...

[2021.03.15 16:19:38] This is the only instance of Telegram, starting server and app...

[2021.03.15 16:19:38] Moved logging from 'E:/Telegram Desktop/log_start1.txt' to 'E:/Telegram Desktop/log.txt'!

[2021.03.15 16:19:38] Old start log 'log_start0.txt' found, deleted: [TRUE]

[2021.03.15 16:19:38] Primary screen DPI: 96

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAOpenSansRegular.ttf' loaded 'DAOpenSansRegular'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAOpenSansRegularItalic.ttf' loaded 'DAOpenSansRegularItalic'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'

[2021.03.15 16:19:38] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'

[2021.03.15 16:19:39] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'

[2021.03.15 16:19:39] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'

[2021.03.15 16:19:39] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'

[2021.03.15 16:19:39] App Info: reading settings...

[2021.03.15 16:19:39] App Info: reading encrypted settings...

[2021.03.15 16:19:39] Lang Info: Loaded cached, keys: 3069

[2021.03.15 16:19:39] OpenAL Logging Level: (not set)

[2021.03.15 16:19:39] Audio Playback Devices: OpenAL Soft on Speakers (USB Audio Device);OpenAL Soft on VA2719 Series (NVIDIA High Definition Audio)

[2021.03.15 16:19:39] Audio Playback Default Device: OpenAL Soft on Speakers (USB Audio Device)

[2021.03.15 16:19:39] Audio Capture Devices: OpenAL Soft on Microphone (USB Audio Device)

[2021.03.15 16:19:39] Audio Capture Default Device: OpenAL Soft on Microphone (USB Audio Device)

[2021.03.15 16:19:39] App Info: reading accounts info...

[2021.03.15 16:19:39] App Info: bad decrypt key, data not decrypted - incorrect password?

[2021.03.15 16:19:39] App Info: could not decrypt pass-protected key from info file, maybe bad password...

[2021.03.15 16:19:41] App Info: reading accounts info...

[2021.03.15 16:19:41] App Info: reading encrypted info...

[2021.03.15 16:19:41] App Info: reading map...

[2021.03.15 16:19:41] App Info: reading encrypted map...

[2021.03.15 16:19:41] App Info: reading encrypted user settings...

[2021.03.15 16:19:41] App Info: encrypted user settings read.

[2021.03.15 16:19:41] App Info: reading encrypted mtp data...

[2021.03.15 16:19:41] MTP Info: read keys, current: 5, to destroy: 0

[2021.03.15 16:19:41] Map read time: 4

[2021.03.15 16:19:41] App Info: reading encrypted mtp config...

[2021.03.15 16:19:42] Export Info: Destroy top bar by controller removal.

[2021.03.15 16:19:47] API Warning: not loaded minimal channel applied.

[2021.03.15 16:24:51] FINISHED UPDATE FOR: 1

[2021.03.15 16:24:51] LOADER ID: 413830552

[2021.03.15 16:24:51] SENT
</details>

**Comment by ilya-fedin** [comment]
Seem to be a duplicate of #5518, but was closed by reporter for some reason.
TL;DR someone interested should add RTL support to https://github.com/desktop-app/lib_ui/blob/master/ui/text/text.cpp (this is a copy of QText from Qt 5.6, that didn't have proper RTL support).

**Comment by Muaath5** [comment]
> 
> 
> Seem to be a duplicate of #5518, but was closed by reporter for some reason.
> TL;DR someone interested should add RTL support to https://github.com/desktop-app/lib_ui/blob/master/ui/text/text.cpp (this is a copy of QText from Qt 5.6 that didn't have proper RTL support).

Ok, Can you add label: "3rd party issue" please?

**Comment by ilya-fedin** [comment]
> Ok, Can you add label: "3rd party issue" please?

This is not a 3rd party issue, lib_ui is a part of tdesktop

**Comment by Muaath5** [comment]
[Linked issue on bugs & suggestions platform](https://bugs.telegram.org/c/3020)

**Comment by Muaath5** [comment]
No, this problem wasn't fixed

**Comment by Muaath5** [comment]
The problem wan't fixed yet

**Comment by mehdikhody** [comment]
I have the same problem in Farsi paragraphs. in android app everything is displayed correctly but in desktop app English words and numbers in Farsi paragraphs are reversed.
